Cannot use `@deephaven/components` in React v18

In a React v18 project, if you try and install `@deephaven/components`, you get an error:
```sh
✗ npm install @deephaven/components
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: demo-day@0.0.0
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"^18.2.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^17.x" from @deephaven/components@0.68.0
npm ERR! node_modules/@deephaven/components
npm ERR!   @deephaven/components@"*" from the root project
```

We should allow support for more versions of React as a peer dependency, like other component libraries such as Spectrum do: https://github.com/adobe/react-spectrum/blob/89fa370bd1bb346be643c9ab2a0d79fab857e4d7/packages/%40react-spectrum/toast/package.json#L57

--- a/packages/app-utils/package.json
+++ b/packages/app-utils/package.json
@@ -51,8 +51,8 @@
     "fira": "mozilla/fira#4.202"
   },
   "peerDependencies": {
-    "react": "^17.x",
-    "react-dom": "^17.x",
+    "react": ">=16.8.0",
+    "react-dom": ">=16.8.0",
     "react-redux": "^7.x",
     "redux": "^4.x"
   },

--- a/packages/auth-plugins/package.json
+++ b/packages/auth-plugins/package.json
@@ -49,8 +49,8 @@
     "@types/react": "^17.0.2"
   },
   "peerDependencies": {
-    "react": "^17.x",
-    "react-dom": "^17.x"
+    "react": ">=16.8.0",
+    "react-dom": ">=16.8.0"
   },
   "files": [
     "dist"

--- a/packages/chart/package.json
+++ b/packages/chart/package.json
@@ -44,7 +44,7 @@
     "react-plotly.js": "^2.6.0"
   },
   "peerDependencies": {
-    "react": "^17.x"
+    "react": ">=16.8.0"
   },
   "devDependencies": {
     "@deephaven/jsapi-shim": "file:../jsapi-shim",

--- a/packages/components/package.json
+++ b/packages/components/package.json
@@ -49,8 +49,8 @@
     "shortid": "^2.2.16"
   },
   "peerDependencies": {
-    "react": "^17.x",
-    "react-dom": "^17.x"
+    "react": ">=16.8.0",
+    "react-dom": ">=16.8.0"
   },
   "devDependencies": {
     "@deephaven/mocks": "file:../mocks"

--- a/packages/console/package.json
+++ b/packages/console/package.json
@@ -47,8 +47,8 @@
     "shortid": "^2.2.16"
   },
   "peerDependencies": {
-    "react": "^17.x",
-    "react-dom": "^17.x"
+    "react": ">=16.8.0",
+    "react-dom": ">=16.8.0"
   },
   "devDependencies": {
     "@deephaven/jsapi-shim": "file:../jsapi-shim",

--- a/packages/dashboard-core-plugins/package.json
+++ b/packages/dashboard-core-plugins/package.json
@@ -61,8 +61,8 @@
     "shortid": "^2.2.16"
   },
   "peerDependencies": {
-    "react": "^17.0.0",
-    "react-dom": "^17.0.0",
+    "react": ">=16.8.0",
+    "react-dom": ">=16.8.0",
     "react-redux": "^7.2.4"
   },
   "devDependencies": {

--- a/packages/dashboard/package.json
+++ b/packages/dashboard/package.json
@@ -35,9 +35,9 @@
     "shortid": "^2.2.16"
   },
   "peerDependencies": {
-    "react": "^17.0.0",
-    "react-dom": "^17.0.0",
-    "react-is": "^17.0.0",
+    "react": ">=16.8.0",
+    "react-dom": ">=16.8.0",
+    "react-is": ">=16.8.0",
     "react-redux": "^7.2.4"
   },
   "devDependencies": {

--- a/packages/file-explorer/package.json
+++ b/packages/file-explorer/package.json
@@ -34,7 +34,7 @@
     "prop-types": "^15.7.2"
   },
   "peerDependencies": {
-    "react": "^17.0.0"
+    "react": ">=16.8.0"
   },
   "devDependencies": {
     "@deephaven/mocks": "file:../mocks"

--- a/packages/golden-layout/package.json
+++ b/packages/golden-layout/package.json
@@ -14,8 +14,8 @@
     "jquery": "^3.6.0"
   },
   "peerDependencies": {
-    "react": "^17.x",
-    "react-dom": "^17.x"
+    "react": ">=16.8.0",
+    "react-dom": ">=16.8.0"
   },
   "scripts": {
     "build": "cross-env NODE_ENV=production run-p build:*",

--- a/packages/grid/package.json
+++ b/packages/grid/package.json
@@ -22,7 +22,7 @@
     "build:sass": "sass --embed-sources --load-path=../../node_modules ./src:./dist"
   },
   "peerDependencies": {
-    "react": "^17.x"
+    "react": ">=16.8.0"
   },
   "files": [
     "dist"

--- a/packages/iris-grid/package.json
+++ b/packages/iris-grid/package.json
@@ -61,8 +61,8 @@
     "shortid": "^2.2.16"
   },
   "peerDependencies": {
-    "react": "^17.x",
-    "react-dom": "^17.x"
+    "react": ">=16.8.0",
+    "react-dom": ">=16.8.0"
   },
   "devDependencies": {
     "@deephaven/jsapi-shim": "file:../jsapi-shim",

--- a/packages/jsapi-bootstrap/package.json
+++ b/packages/jsapi-bootstrap/package.json
@@ -31,7 +31,7 @@
     "react": "^17.x"
   },
   "peerDependencies": {
-    "react": "^17.x"
+    "react": ">=16.8.0"
   },
   "files": [
     "dist"

--- a/packages/jsapi-components/package.json
+++ b/packages/jsapi-components/package.json
@@ -40,7 +40,7 @@
     "react-test-renderer": "^17.0.2"
   },
   "peerDependencies": {
-    "react": "^17.x"
+    "react": ">=16.8.0"
   },
   "files": [
     "dist"

--- a/packages/plugin/package.json
+++ b/packages/plugin/package.json
@@ -33,7 +33,7 @@
     "@fortawesome/react-fontawesome": "^0.2.0"
   },
   "peerDependencies": {
-    "react": "^17.x"
+    "react": ">=16.8.0"
   },
   "files": [
     "dist"

--- a/packages/pouch-storage/package.json
+++ b/packages/pouch-storage/package.json
@@ -31,7 +31,7 @@
     "pouchdb-find": "^7.3.0"
   },
   "peerDependencies": {
-    "react": "^17.x"
+    "react": ">=16.8.0"
   },
   "files": [
     "dist"

--- a/packages/react-hooks/package.json
+++ b/packages/react-hooks/package.json
@@ -28,7 +28,7 @@
     "shortid": "^2.2.16"
   },
   "peerDependencies": {
-    "react": "^17.x"
+    "react": ">=16.8.0"
   },
   "files": [
     "dist"

--- a/packages/storage/package.json
+++ b/packages/storage/package.json
@@ -26,7 +26,7 @@
     "lodash.throttle": "^4.1.1"
   },
   "peerDependencies": {
-    "react": "^17.x"
+    "react": ">=16.8.0"
   },
   "files": [
     "dist"

